core/src/transaction_meta.rs

```diff
@@ -22,36 +22,47 @@ impl From<BitVecSerde> for BitVec {
 pub struct TransactionMeta {
     block_number: u64,
     epoch_number: u64,
-    /// first bit indicate if transaction is a cellbase transaction
-    /// next bits indicate if transaction has dead cells
+    cellbase: bool,
+    /// each bits indicate if transaction has dead cells
     #[serde(with = "BitVecSerde")]
     dead_cell: BitVec,
 }
 
 impl TransactionMeta {
-    pub fn new(block_number: u64, epoch_number: u64, outputs_count: usize) -> TransactionMeta {
+    pub fn new(
+        block_number: u64,
+        epoch_number: u64,
+        outputs_count: usize,
+        all_dead: bool,
+    ) -> TransactionMeta {
         TransactionMeta {
             block_number,
             epoch_number,
-            dead_cell: BitVec::from_elem(outputs_count + 1, false),
+            cellbase: false,
+            dead_cell: BitVec::from_elem(outputs_count, all_dead),
         }
     }
 
     /// New cellbase transaction
-    pub fn new_cellbase(block_number: u64, epoch_number: u64, outputs: usize) -> Self {
-        let mut result = Self::new(block_number, epoch_number, outputs);
-        result.dead_cell.set(0, true);
+    pub fn new_cellbase(
+        block_number: u64,
+        epoch_number: u64,
+        outputs: usize,
```

Comment:
inconsistent naming in fn `new`
```suggestion
        outputs_count: usize,
```